src/Java.Interop/Java.Interop/JniRuntime.JniTypeManager.cs

```diff
@@ -43,7 +45,54 @@ public JniTypeSignature GetTypeSignature (Type type)
 			{
 				AssertValid ();
 
-				return GetTypeSignatures (type).FirstOrDefault ();
+				if (type == null)
+					throw new ArgumentNullException (nameof (type));
+				if (type.ContainsGenericParameters)
+					throw new ArgumentException ("Generic type definitions are not supported.", nameof (type));
+
+				type = GetUnderlyingType (type, out int rank);
+#if !XA_INTEGRATION
```

Comment:
We don't actually need to keep the `XA_INTEGRATION` defines around, as that hasn't been meaningful since xamarin/xamarin-android@7d32ef3e6ef88eba004f1d2a9f9eb0f42cbf887e and the removal of the `XAIntegrationDebug` and `XAIntegrationRelease` configurations in commit 80b4f4eb578283a1eefa1f476474dccb15769e5b.